src/e131.c

```diff
@@ -164,6 +164,28 @@ int e131_multicast_join_iface(int sockfd, const uint16_t universe, const int ifi
   return setsockopt(sockfd, IPPROTO_IP, IP_ADD_MEMBERSHIP, &mreq, sizeof mreq);
 }
 
+/* Join a socket file descriptor to an E1.31 multicast group using a universe and an IP address to bind to */
+extern int e131_multicast_join_ifaddr(int sockfd, const uint16_t universe, const char *ifaddr) {
+    if (universe < 1 || universe > 63999) {
+        errno = EINVAL;
+        return -1;
+    }
+#ifdef _WIN32
+    if (ifindex != 0) {
+    errno = ENOSYS;
+    return -1;
+  }
+  struct ip_mreq mreq;
+  mreq.imr_multiaddr.s_addr = htonl(0xefff0000 | universe);
+  mreq.imr_interface.s_addr = inet_addr(ifaddr);
+#else
+    struct ip_mreqn mreq;
+    mreq.imr_multiaddr.s_addr = htonl(0xefff0000 | universe);
+    mreq.imr_address.s_addr = inet_addr(ifaddr);
+#endif
+    return setsockopt(sockfd, IPPROTO_IP, IP_ADD_MEMBERSHIP, &mreq, sizeof mreq);
+}
+
```

Comment:
Just two minor comments:
* The indentation is not consistent with the rest of the code, which uses two spaces.
* Can you explicitly set `mreq.imr_ifindex` to `0` on each codepath? I prefer to be explicit and according to the documentation: _imr_ifindex is the interface index of the interface that should join/leave the imr_multiaddr group, or 0 to indicate any interface._

Reply:
@hhromic Thanks for the feedback! I've updated the spacing as you requested, and set `mreq.imr_ifindex` to `0` on each code path.

Reply:
Thanks for addressing the comments, appreciated!
I just realized another issue, will comment in another separate comment.